plugins/wazuh-fleet/public/components/agents/list/list.tsx

```diff
@@ -3,126 +3,182 @@ import {
   EuiPageHeader,
   EuiSpacer,
   EuiButton,
-  EuiPopover,
-  EuiContextMenuPanel,
-  EuiContextMenuItem,
-  EuiHorizontalRule,
   EuiFlyout,
   EuiFlyoutBody,
   EuiFlyoutHeader,
   EuiText,
   EuiTitle,
   EuiLink,
 } from '@elastic/eui';
+import { Agent, IAgentResponse } from '../../../../common/types';
+import { AgentResume } from '../details/resume';
+import { getAgentManagement, getCore } from '../../../plugin-services';
+import { NavigationService } from '../../../react-services/navigation-service';
+import { enrollmentAgent } from '../../common/views';
+import { TableIndexer } from '../../common/table-indexer/table-indexer';
+import { ConfirmModal } from '../../common/confirm-modal/confirm-modal';
 import { agentsTableColumns } from './columns';
 import { AgentsVisualizations } from './visualizations';
-import { Agent } from '../../../../common/types';
-import { AgentResume } from '../details/resume';
-import { getCore } from '../../../plugin-services';
+import { EditAgentGroupsModal } from './actions/edit-groups-modal';
+import { UpgradeAgentModal } from './actions/upgrade-agent-modal';
+import { AgentsTableGlobalActions } from './global-actions/global-actions';
+import { EditAgentNameModal } from './actions/edit-name-agent-modal';
 
 export interface AgentListProps {
-  FleetDataSource: any;
-  FleetDataSourceRepository: any;
-  TableIndexer: any;
+  indexPatterns: any;
+  filters: any[];
```

Comment:
The filters prop is typed as any. Using any defeats the purpose of TypeScript. Consider defining a specific type for filters to improve type safety and maintainability.

import { Filter } from '../../../../src/plugins/data/public';

```suggestion
  filters: Filter[];
```

Reply:
Done https://github.com/wazuh/wazuh-dashboard-plugins/commit/e20fd9edb79d568a1c9b6a4a68543fc244bd8b11